src/ray/core_worker/transport/raylet_transport.h

```diff
@@ -31,8 +31,8 @@ class CoreWorkerRayletTaskReceiver {
                         rpc::SendReplyCallback send_reply_callback);
 
  private:
-  /// Raylet client.
-  std::unique_ptr<RayletClient> &raylet_client_;
+  /// Raylet client. TODO(ekl) having a shared ptr ref is a hack.
```

Comment:
It turns out the benefit of `unique_ptr&` was that you could initialize it to point to the core_worker member pointer, and initialize the core_worker member later. This works around circular initialization dependency between RayletClient and the task receivers.

Not sure if this is the best solution.